orangecontrib/spectroscopy/widgets/owcos.py

```diff
@@ -442,18 +419,34 @@ def plotCOS(self):
             self.cbarCOS.set_range(-1 * np.nanmax(np.absolute(cosmat)), np.nanmax(np.absolute(cosmat)))
             self.cbarCOS.set_colors(np.array(colorcet.diverging_bwr_40_95_c42) * 255)
 
-            for s in leftSP:
-                self.left_plot.plot(s, leftSPwn, pen=pg.mkPen(color=(50, 50, 50), width=0.5))
+            if len(leftSP) > 100:
+                left_indices = np.linspace(0, len(leftSP)-1, 100, dtype=int)
```

Comment:
Don't generate indices with linspace, it is complex (see the exception) and just not made for it. To me, having to error round to get indices, always smells. Rather use: `random.Random(0).sample(len(whatever), 100)` to generate indices which are only needed if you have too many spectra. There should be no else

Reply:
I am using this because we should do an equidistant sampling of the spectra (usually these experiments describe something changing as a function of time/temperature/etc), random is not a good choice in this case...

Reply:
Ok, I understand your argument. If being equidistant is important, then this is fine.

Random of 100 is almost equidistant though. :)